Mcstasscript tutorial

mads-bertelsen opened the pull request:
Addition of McStasScript tutorial with a single exercise on guide simulation. A solution to the exercise is included.

Commit:
Initial commit of McStasScript tutorial for python course.

Commit:
Changed name of McStas notebooks from 6 to 7 to avoid clashing with scipp.

Commit:
Added introduction and an exercise with guide simulation.

Commit:
Change name of introduction so it appears at top of list

Commit:
Small fixes to McStas_basics

trnielsen commented:
@mads-bertelsen 

I have tested the notebooks on the two containers for the Python workshop. And they work fine. There are only smaller issues we should address. 

1) CPU time and MPI
In some of the Union notebooks the cell execution time is one the order of 30-40 seconds. And especially for Union4 it takes more the 12 minutes to run some of the cells.

In 0_McStas_session_introduction it would be advantageous to highlight this. Something like a  sentence saying "In some of the Union notebooks an increased CPU time is to be expected. Some of the Union notebooks have cell execution time on the order of 30-40 seconds.". The Union4 notebook should also be mentioned, since by use of MPI=4 we can go from 12 minutes to 3 minuets  CPU time. 

2) Imagemagick missing
Union3 notebook section 1.6.1 can not create an gif animation since Imagemagic is not install. However, if I install Imagemagick on a local container the gif can be created. But the then markdown cell ![SegmentLocal](animation_demo_long.gif "Animation") fails. And the gif is not displayed. 

Given the short time to the workshop I suggest that we just add the animation.gif to this repo. 

3) Union7 - wrong file
In Union7 section 1.2 with open("run_folder/Union_history.dat") as file: should be with lower case: with open("run_folder/union_history.dat") as file:
E.i: Union_history.dat -> union_history.dat

Commit:
Removed animation part of logger notebook due to long simulation times
and problems with viewing the resulting gif.

Added mpi=4 to most long simulations as 4 cores are available.

Fixed Union_history -> union_history data name error.

Commit:
Added mention of 4 cores in IKON jupyter instance.

mads-bertelsen commented:
Added mpi=4 to the long runs, decreased N count a bit in critical places.

Removed the animation part of the tutorial, there is simply enough material and these runs are the longest.

Fixed the filename in Union_7.
